src/transforms/filter.js

```diff
@@ -8,211 +8,269 @@
 
 'use strict';
 
-var isNumeric = require('fast-isnumeric');
-
-// var Lib = require('@src/lib');
 var Lib = require('../lib');
+var axisIds = require('../plots/cartesian/axis_ids');
 
-/* eslint no-unused-vars: 0*/
-
-// so that Plotly.register knows what to do with it
 exports.moduleType = 'transform';
 
-// determines to link between transform type and transform module
 exports.name = 'filter';
 
-// ... as trace attributes
 exports.attributes = {
+    active: {
+        valType: 'boolean',
+        dflt: true,
+        description: [
+            'Toggles whether or not the filter is active.'
+        ].join(' ')
+    },
+    filtersrc: {
+        valType: 'enumerated',
+        values: ['x', 'y', 'z', 'ids'],
+        dflt: 'x',
+        description: [
+            'Sets the variable which the filter will be applied.'
+        ].join(' ')
+    },
     operation: {
         valType: 'enumerated',
         values: ['=', '<', '>', 'within', 'notwithin', 'in', 'notin'],
-        dflt: '='
+        dflt: '=',
+        description: [
+            'Sets the filter operation.'
+        ].join(' ')
     },
     value: {
         valType: 'any',
-        dflt: 0
+        dflt: 0,
+        description: [
+            'Sets the value or values by which to filter by.',
+            'Values are expected to be in the same type as the data linked',
+            'to *filtersrc*.',
+            'When `operation` is set to *within* and *notwithin*',
+            '*value* is expected to be 2-item array where the first item',
+            'is the lower bound and the second item is the upper bound.',
+            'When `operation`, is set to *in*, *notin* '
+        ].join(' ')
     },
-    filtersrc: {
-        valType: 'enumerated',
-        values: ['x', 'y', 'ids'],
-        dflt: 'x',
-        ids: {
-            valType: 'data_array',
-            description: 'A list of keys for object constancy of data points during animation'
-        }
+    strictinterval: {
+        valType: 'boolean',
+        dflt: true,
+        arrayOk: true,
+        description: [
+            'Determines whether or not the filter operation includes data item value,',
+            'equal to *value*.',
+            'Has only an effect for `operation` *>*, *<*, *within* and *notwithin*',
+            'When `operation` is set to  *within* and *notwithin*,',
+            '`strictinterval` is expected to be a 2-item array where the first (second)',
+            'item determines strictness for the lower (second) bound.'
```

Comment:
Yeah my main concern was if we use `[]`, `()`, `[)`, and `(]` instead of `within` what would happen to the `notwithin` option?
I was thinking a similar thing to what @alexcjohnson suggested - if we go with `[]`, `()`, `[)`, and `(]` for `within` I think there should be a parallel syntax for `notwithin`. I don't think `][`, `](`, `)[`, and `)(` is *too* confusing (as long as we write a thorough description).